cmd/kube-apiserver/app/options/options.go

```diff
@@ -121,39 +122,53 @@ func NewAPIServer() *APIServer {
 }
 
 // dest must be a map of group to groupVersion.
-func gvToMap(gvList string, dest map[string]string) {
-	for _, gv := range strings.Split(gvList, ",") {
-		if gv == "" {
+func mergeGroupVersionIntoMap(gvList string, dest map[string]unversioned.GroupVersion) error {
+	for _, gvString := range strings.Split(gvList, ",") {
+		if gvString == "" {
 			continue
 		}
 		// We accept two formats. "group/version" OR
 		// "group=group/version". The latter is used when types
 		// move between groups.
-		if !strings.Contains(gv, "=") {
-			dest[apiutil.GetGroup(gv)] = gv
+		if !strings.Contains(gvString, "=") {
+			gv, err := unversioned.ParseGroupVersion(gvString)
+			if err != nil {
+				return err
+			}
+			dest[gv.Group] = gv
+
 		} else {
-			parts := strings.SplitN(gv, "=", 2)
-			// TODO: error checking.
-			dest[parts[0]] = parts[1]
+			parts := strings.SplitN(gvString, "=", 2)
+			gv, err := unversioned.ParseGroupVersion(parts[1])
+			if err != nil {
+				return err
+			}
+			dest[parts[0]] = gv
 		}
 	}
+
+	return nil
 }
 
-// StorageGroupsToGroupVersions returns a map from group name to group version,
+// StorageGroupsEncodingVersionOverrides returns a map from group name to group version,
 // computed from the s.DeprecatedStorageVersion and s.StorageVersions flags.
 // TODO: can we move the whole storage version concept to the generic apiserver?
-func (s *APIServer) StorageGroupsToGroupVersions() map[string]string {
-	storageVersionMap := map[string]string{}
+func (s *APIServer) StorageGroupsEncodingVersionOverrides() (map[string]unversioned.GroupVersion, error) {
```

Comment:
> Strictly speaking, this gives all the storage group versions and not just the overrides (since this also merges DefaultStorageVersions)

Changed to `StorageGroupsToEncodingVersion`